src/test/chainstate_tests.cpp

```diff
@@ -0,0 +1,602 @@
+    BOOST_CHECK(comparator(&indexB, &indexA) == true);  // B < A (higher seq ID, so B comes before A)
+
+    // Case 2: Same height, same sequence ID (pointer tie-breaker)
+    // This happens with blocks loaded from disk (both have id = 0)
+    indexA.nHeight = 100;
+    indexA.nSequenceId = 0;
+    indexB.nHeight = 100;
+    indexB.nSequenceId = 0;
+
+    // With same height and sequence ID, pointer address determines order
+    bool resultAB = comparator(&indexA, &indexB);
+    bool resultBA = comparator(&indexB, &indexA);
+
+    // They should give opposite results (unless indexA == indexB)
+    if (&indexA != &indexB) {
+        BOOST_CHECK(resultAB != resultBA);
+    }
+
+    // Case 3: Identical blocks (same pointer)
+    BOOST_CHECK(comparator(&indexA, &indexA) == false);
+
+    // Case 4: Height difference takes precedence
+    indexA.nHeight = 99;
+    indexA.nSequenceId = 100;
+    indexB.nHeight = 100;
+    indexB.nSequenceId = 1;
+
+    BOOST_CHECK(comparator(&indexA, &indexB) == true);  // Lower height comes first
+
+    // Case 5: Maximum sequence ID edge case
+    indexA.nHeight = 100;
+    indexA.nSequenceId = std::numeric_limits<int32_t>::max();
+    indexB.nHeight = 100;
+    indexB.nSequenceId = std::numeric_limits<int32_t>::max() - 1;
+
+    BOOST_CHECK(comparator(&indexA, &indexB) == true);  // Higher seq ID (max) comes first
+    BOOST_CHECK(comparator(&indexB, &indexA) == false); // Lower seq ID comes after
+
+    // Case 6: Minimum sequence ID edge case (used in PreciousBlock)
+    indexA.nHeight = 100;
+    indexA.nSequenceId = std::numeric_limits<int32_t>::min();
+    indexB.nHeight = 100;
+    indexB.nSequenceId = std::numeric_limits<int32_t>::min() + 1;
+
+    BOOST_CHECK(comparator(&indexA, &indexB) == false); // Lower (min) seq ID comes after
+    BOOST_CHECK(comparator(&indexB, &indexA) == true);  // Higher seq ID comes first
+}
+
+/**
+ * Test BlockHasher edge cases
+ *
+ * BlockHasher is used in the BlockMap (unordered_map). We test that it
+ * properly computes hashes and handles edge cases like zero hashes.
+ */
+BOOST_AUTO_TEST_CASE(block_hasher_edge_cases)
+{
+    BlockHasher hasher;
+
+    // Case 1: Zero hash
+    uint256 zeroHash;
+    zeroHash.SetNull();
+    size_t hash1 = hasher(zeroHash);
+    BOOST_CHECK(hash1 == 0); // Cheap hash of zero should be 0
+
+    // Case 2: Same hash should produce same value
+    uint256 hash256a = InsecureRand256();
+    size_t hashValueA1 = hasher(hash256a);
+    size_t hashValueA2 = hasher(hash256a);
+    BOOST_CHECK(hashValueA1 == hashValueA2);
+
+    // Case 3: Different hashes should (likely) produce different values
+    uint256 hash256b = InsecureRand256();
+    size_t hashValueB = hasher(hash256b);
+
+    // While hash collisions are possible, they should be rare
+    if (hash256a != hash256b) {
+        // Just verify the hasher runs; collision probability is low but non-zero
+        BOOST_CHECK(hashValueA1 != hashValueB || hashValueA1 == hashValueB);
```

Comment:
This check is not a test because it will always be true:

A != B || A == B → always true